src/scripts/main.js

```diff
@@ -1,3 +1,280 @@
+  headRows.sort((rowA, rowB) => {
+    const cellA = rowA.cells[index].textContent.trim();
+    const cellB = rowB.cells[index].textContent.trim();
+
+    const direction = sortDirections[index] ? 1 : -1;
+
+    return compareValues(cellA, cellB, index) * direction;
+  });
+
+  tableBody.append(...headRows);
+  lastSortedIndex = index;
+}
+
+function compareValues(cellA, cellB, index) {
+  if (index <= 2) {
+    return cellA.localeCompare(cellB);
+  }
+
+  if (index === 4) {
+    const valueA = parseFloat(cellA.replace(/[$,]/g, '')) || 0;
+    const valueB = parseFloat(cellB.replace(/[$,]/g, '')) || 0;
+
+    return valueA - valueB;
+  }
+
+  return parseFloat(cellA) - parseFloat(cellB);
+}
+// #endregion
+
+// #region selected Row
+const rows = document.querySelectorAll('tbody tr');
+let activeRow = null;
+
+rows.forEach((row, index) => {
+  row.addEventListener('click', () => selectRow(row));
+});
+
+function selectRow(row) {
+  if (activeRow) {
+    activeRow.classList.remove('active');
+  }
+
+  row.classList.add('active');
+
+  activeRow = row;
+}
+// #endregion
+
+// #region form to add new employee
+const form = document.createElement('form');
+
+form.classList.add('new-employee-form');
+
+const fields = [
+  {
+    label: 'Name:',
+    name: 'name',
+    type: 'text',
+    data: 'name',
+  },
+  {
+    label: 'Position:',
+    name: 'position',
+    type: 'text',
+    data: 'position',
+  },
+  {
+    label: 'Age:',
+    name: 'age',
+    type: 'number',
+    data: 'age',
+  },
+  {
+    label: 'Salary:',
+    name: 'salary',
+    type: 'number',
+    data: 'salary',
+  },
+];
+
+const selectField = {
+  label: 'Office:',
+  name: 'office',
+  options: [
+    { value: 'tokyo', text: 'Tokyo' },
+    { value: 'singapore', text: 'Singapore' },
+    { value: 'london', text: 'London' },
+    { value: 'new-York', text: 'New York' },
+    { value: 'edinburgh', text: 'Edinburgh' },
+    { value: 'san-Francisco', text: 'San Francisco' },
+  ],
+  data: 'office',
+};
+
+fields.forEach((field, index) => {
+  const label = document.createElement('label');
+
+  label.textContent = field.label;
+
+  const input = document.createElement('input');
+
+  input.type = field.type;
+  input.name = field.name;
+  input.setAttribute('data-qa', field.data);
+  input.required = true;
+
+  label.appendChild(input);
+  form.appendChild(label);
+
+  if (index === 1) {
+    const selectLabel = document.createElement('label');
+    const select = document.createElement('select');
+
+    selectLabel.textContent = selectField.label;
+    select.name = selectField.name;
+    select.setAttribute('data-qa', selectField.data);
+    // select.required = true;
+
+    selectField.options.forEach((optionData) => {
+      const option = document.createElement('option');
+
+      option.value = optionData.value;
+      option.textContent = optionData.text;
+      select.appendChild(option);
+    });
+
+    selectLabel.appendChild(select);
+    form.appendChild(selectLabel);
+  }
```

Comment:
The logic for adding a select field is embedded within the loop for input fields. Consider separating this logic to improve readability and maintainability.